Public/Src/FrontEnd/Script/Evaluator/Converter.cs

```diff
@@ -665,6 +665,61 @@ public static string ExtractStringLiteral(ObjectLiteral literal, SymbolAtom prop
             return ExpectNumber(value, context: new ConversionContext(name: property, objectCtx: literal));
         }
 
+        /// <summary>
+        /// Extracts optional boolean or int property.
+        /// Returns 'false' if the property is missing in a given object literal
+        /// Returns 'true' if the property is either bool or int.
+        /// </summary>
+        /// <remarks>
+        /// Either enabled or invValue will contain value not both. 
+        /// </remarks>
+        public static bool ExtractOptionalBooleanOrInt(ObjectLiteral literal, SymbolAtom property, out bool? enabled, out int? intValue)
+        {
+            enabled = false;
+            intValue = 0;
+
+            var value = literal[property];
+            if (value.IsUndefined)
+            {
+                return false;
+            }
+
+            if(!TryGetBooleanOrInt(value, out enabled, out intValue))
+            {
+                throw CreateException(
+                   new[] { typeof(bool), typeof(int) },
+                   value,
+                   context: new ConversionContext(name: property, objectCtx: literal));
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Try to get int or bool value from EvaluationResult.
+        /// Returns 'true' if value is int or bool type, otherwise return 'false'
+        /// </summary>
+        public static bool TryGetBooleanOrInt(EvaluationResult value, out bool? boolValue, out int? intValue)
+        {
+            boolValue = default(bool);
+            intValue = default(int);

```

Comment:
This should be `default` instead of `default(type)`

Reply:
fixed in new iteration